packages/web3-eth/src/types.ts

```diff
@@ -0,0 +1,194 @@
+import { AccessList, Log, TransactionHash, TransactionInfo, Uncles } from 'web3-common';
+import {
+	Address,
+	HexString,
+	Numbers,
+	ValidTypes,
+	ValidReturnTypes,
+	HexString32Bytes,
+	HexString256Bytes,
+	HexStringBytes,
+	Uint,
+} from 'web3-utils';
+
+export enum ChainNames {
+	MAINNET = 'mainnet',
+	GOERLI = 'goerli',
+	KOVAN = 'kovan',
+	RINKEBY = 'rinkeby',
+	ROPSTEN = 'ropsten',
+}
+
+export enum HardForks {
+	CHAIN_START = 'chainstart',
+	HOMESTEAD = 'homestead',
+	DAO = 'dao',
+	TANGERINE_WHISTLE = 'tangerineWhistle',
+	SPURIOUS_DRAGON = 'spuriousDragon',
+	BYZANTIUM = 'byzantium',
+	CONSTANTINOPLE = 'constantinople',
+	PETERSBURG = 'petersburg',
+	ISTANBUL = 'istanbul',
+	BERLIN = 'berlin',
+	LONDON = 'london',
+}
+
+export interface Transaction {
+	from?: Address;
+	to?: Address;
+	value?: Numbers;
+	gas?: Numbers;
+	gasPrice?: Numbers;
+	type?: Numbers;
+	maxFeePerGas?: Numbers;
+	maxPriorityFeePerGas?: Numbers;
+	accessList?: AccessList;
+	input: HexStringBytes;
+	data?: HexString;
+	nonce?: Numbers;
+	chain?: HexString;
+	hardfork?: HexString;
+	common?: {
+		customChain: {
+			name?: string;
+			networkId: Numbers;
+			chainId: Numbers;
+		};
+		baseChain?: ChainNames;
+		hardfork?: HardForks;
+	};
+}
+
+interface BaseTransactionFormatted<
+	DesiredType extends ValidTypes = ValidTypes.HexString,
+	ReturnType = ValidReturnTypes[DesiredType],
+> {
+	readonly to?: Address | null;
+	readonly type: ReturnType;
+	readonly nonce: ReturnType;
+	readonly gas: ReturnType;
+	readonly value: ReturnType;
+	readonly input: HexStringBytes;
+}
+
+interface Transaction1559UnsignedFormatted<
+	DesiredType extends ValidTypes = ValidTypes.HexString,
+	ReturnType = ValidReturnTypes[DesiredType],
+> extends BaseTransactionFormatted<DesiredType> {
+	readonly maxFeePerGas: ReturnType;
+	readonly maxPriorityFeePerGas: ReturnType;
+	readonly accessList: AccessList;
+}
```

Comment:
We can simply all these types with better extended generics. Please create an issue just stating to simply all formatted types. So we address it later.

Reply:
Created an [issue](https://github.com/ChainSafe/web3.js/issues/4632) for this, and added it to [Web3Eth TODO list](https://github.com/ChainSafe/web3.js/issues/4608)